test/e2e/framework.go

```diff
@@ -704,15 +712,78 @@ func (data *TestData) enableAntreaFlowExporter(ipfixCollector string) error {
 	return data.mutateAntreaConfigMap(nil, ac, false, true)
 }
 
-func (data *TestData) disableAntreaFlowExporter() error {
-	ac := func(config *agentconfig.AgentConfig) {
-		config.FeatureGates["FlowExporter"] = false
+// deployFlowVisibilityClickHouse deploys ClickHouse operator and DB.
+func (data *TestData) deployFlowVisibilityClickHouse() (*PodIPs, error) {
+	err := data.CreateNamespace(flowVisibilityNamespace, nil)
+	if err != nil {
+		return nil, err
 	}
-	return data.mutateAntreaConfigMap(nil, ac, false, true)
+
+	if err := wait.Poll(2*time.Second, 10*time.Second, func() (bool, error) {
+		rc, stdout, stderr, err := data.provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply -f %s", flowVisibilityYML))
+		if err != nil || rc != 0 {
+			// ClickHouseInstallation CRD from ClickHouse Operator install bundle applied soon before
+			// applying CR. Sometimes apiserver validation fails to recognize resource of
+			// kind: ClickHouseInstallation. Retry in such scenario.
+			if strings.Contains(stderr, "ClickHouseInstallation") || strings.Contains(stdout, "ClickHouseInstallation") {
+				return false, nil
+			}
+			return false, fmt.Errorf("error when deploying the flow visibility YML %s: %s, %s, %v", flowVisibilityYML, stdout, stderr, err)
+		}
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+
+	// check for clickhouse operator running
+	chOperator, err := data.getClickHouseOperator()
+	if err != nil {
+		return nil, fmt.Errorf("error when getting clickhouse-operator Pod: %v", err)
+	}
+	podName := chOperator.Name
+	_, err = data.PodWaitFor(defaultTimeout, podName, kubeNamespace, func(p *corev1.Pod) (bool, error) {
+		for _, condition := range p.Status.Conditions {
+			if condition.Type == corev1.PodReady {
+				return condition.Status == corev1.ConditionTrue, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil {
+		_, stdout, stderr, podErr := data.provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl get po %s -n %s -o yaml", podName, kubeNamespace))
+		return nil, fmt.Errorf("error when waiting for ClickHouse Operator Ready: %v; stdout: %s, stderr: %s, %v", err, stdout, stderr, podErr)
+	}
+	// check for clickhouse pod running
+	_, err = data.PodWaitFor(defaultTimeout, flowVisibilityCHPodName, flowVisibilityNamespace, func(p *corev1.Pod) (bool, error) {
+		for _, condition := range p.Status.Conditions {
+			if condition.Type == corev1.PodReady {
+				return condition.Status == corev1.ConditionTrue, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil {
+		_, stdout, stderr, podErr := data.provider.RunCommandOnNode(controlPlaneNodeName(),
+			fmt.Sprintf("kubectl get po %s -n %s -o yaml", flowVisibilityCHPodName, flowVisibilityNamespace))
+		return nil, fmt.Errorf("error when waiting for ClickHouse Ready: %v; stdout: %s; stderr: %s; %v", err, stdout, stderr, podErr)
+	}
```

Comment:
ClickHouse pod can be a bit slow to become ready, and we want to make sure collectors are running before starting FA to avoid any potential underlying connection retry delay between FA and the collectors. If we're don't wait for collector readiness we'll need to insert static wait between deploying FA and starting the traffic test, or the test could fail as the test is started before FA could even connect to collectors. 

By `tests will become long` do you mean the time cost or the length of code? I still feel it makes sense to at least check CH readiness before starting FA.

Reply:
I don't think that answers my question. This seems to be strictly for debugging purposes since it is conditioned by `err != nil`. I am not talking about the call to `data.PodWaitFor`.

Reply:
got it. removed all 3 occurrences. now just returns err if raised from the wait.